REQUIRE

```diff
@@ -1,4 +1,6 @@
 julia 0.7-beta
 DataFrames 0.11.5
+DataStreams 0.3.0
```

Comment:
Should this use Tables.jl once it ready?

Reply:
This would be an awesome case for Tables.jl; it should be much simpler and I'm happy to help set that up or help in any way.

Reply:
That would be awesome; I was reluctant to fully drop dataframes (even though it's not required for the model matrix stuff anymore) because I didn't want to mess around too much with the `DataFrame(Regression|Statistical)Model` interface in this PR, but it might be the right place to do it if we're all on board.

Reply:
I am willing to put some extra work on my package to see this come through.

Reply:
@quinnj I'm looking at this again; the one point where I'm a little unclear on how to proceed is with the `model_cols` methods.  Currently these dispatch on the second argument as `NamedTuple`, with most handling both a single row and a whole table using broadcast and some having specialized methods for a named tuple of scalars (a row) vs. a named tuple of vectors (an entire table).  I guess where I'm not clear is whether I need to remove that restriction (and define methods like `model_cols(::Term, data)`), and use `istable` and `columns/rows` to get the columns/elements one by one...or whether it's enough to handle that at a high level somewhere. It's nice to be able to call `model_cols` with anything that's a namedtuple-table and have it "just work" but still be able to specialize on particular table types (both for our purposes and if other packages want to add special methods for their own table types...)

The other thing I'm wondering about is whether there are select/filter type capabilities provided by Tables or whether we'll have to keep using the "custom" `_select` and `_filter` methods that I copied from DataFrames...